src/lib/autocomplete/autocomplete-trigger.ts

```diff
@@ -376,8 +374,13 @@ export class MdAutocompleteTrigger implements ControlValueAccessor, OnDestroy {
    */
   private _subscribeToClosingActions(): Subscription {
     const firstStable = first.call(this._zone.onStable);
-    const optionChanges = map.call(this.autocomplete.options.changes, () =>
-      this._positionStrategy.recalculateLastPosition());
+    const optionChanges = switchMap.call(this.autocomplete.options.changes, () => {
+      this._positionStrategy.recalculateLastPosition();
+
+      // Defer emitting to the stream until the next tick, because changing
+      // bindings in here will cause "changed after checked" errors.
+      return Promise.resolve();
```

Comment:
Mixing switchMap and Promise.resolve doesn't feel quite right. Use `do` and `observeOn(asap)` operators here instead?  It would be more rx idiomatic.